.github/workflows/release.yaml

```diff
@@ -4,73 +4,122 @@ on:
+            echo "skip=false" >> "$GITHUB_OUTPUT"
+          fi
+          echo "VERSION_TAG=${VERSION_TAG}" >> "$GITHUB_OUTPUT"
   release:
     runs-on: ubuntu-latest
+    needs:
+      - build-skip-check
+    if: needs.build-skip-check.outputs.skip == 'false'
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - linux/amd64
+          - linux/arm64
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
         with:
-          fetch-depth: "0"
-
+          fetch-depth: 1
       - name: Configure Git
         run: |
           git config user.name "$GITHUB_ACTOR"
           git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
 
+      - name: Set up QEMU
+        timeout-minutes: 1
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
       - name: Install Helm
-        uses: azure/setup-helm@v1
+        uses: azure/setup-helm@v4
         with:
-          version: v3.7.1
-
-      # TODO: Maintainer of repository to follow:
-      # https://github.com/docker/login-action#google-container-registry-gcr to add credentials so
-      # we can push from github actions
-      # - name: log in to google container registry
-      #   uses: docker/login-action@v1
-      #   with:
-      #     registry: gcr.io
-      #     username: ${{ secrets.DOCKER_USERNAME }}
-      #     password: ${{ secrets.DOCKER_PASSWORD }}
-      
+          version: v3.14.3
       - name: Login to Packages Container registry
-        uses: docker/login-action@v1 
+        uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
-
       - name: Build and Push Spark-Operator Docker Image to github container registry
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          platforms: ${{ matrix.platform }}
+          cache-to: type=gha,mode=max
+          cache-from: type=gha
+          push: true
+          outputs: type=image,name=${{ env.REGISTRY_IMAGE }},push-by-digest=true,name-canonical=true,push=true
+      - name: Export digest
         run: |
-          DOCKER_TAG=$(cat charts/spark-operator-chart/Chart.yaml | grep "appVersion: .*" | cut -c13-)
-          docker build -t gcr.io/spark-operator/spark-operator:${DOCKER_TAG} .
-          echo "Ideally, we'd release the docker container at this point, but the maintainer of this repo needs to approve..."
-          docker tag gcr.io/spark-operator/spark-operator:${DOCKER_TAG} ghcr.io/googlecloudplatform/spark-operator:${DOCKER_TAG}
-          if ! docker pull ghcr.io/googlecloudplatform/spark-operator:${DOCKER_TAG}; then
-            docker push ghcr.io/googlecloudplatform/spark-operator:${DOCKER_TAG}
-          else
-            echo "Spark-Operator Docker Image alredy exists"
-          fi
-
```

Comment:
Is the reason for removing these lines to get the CI to pass? If that's the case, I don't think it should be in this PR since it's not related to Go upgrade. As I was suggesting in another PR: https://github.com/kubeflow/spark-operator/pull/1884#issuecomment-2041159084, we should fix the pipelines first

Reply:
lines were not just simply removed, I've replaced it with multiarch image build + helm chart publishing

Reply:
You are right. I was looking for `docker push` but didn't realize that it was achieved via `docker/build-push-action`